redundant-bmc/src/manager.cpp

```diff
@@ -303,9 +303,65 @@ sdbusplus::async::task<> Manager::method_call(start_failover_t /* unused */,
         throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
     }
 
-    // TODO: Implement the failover
-    lg2::error("Failovers are not implemented yet");
-    throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
+    if (redundancyInterface.role() == Role::Passive)
+    {
+        // TODO: Uncomment: ctx.spawn(doFailoverFromPassive());
+        // And delete these 2 lines
+        lg2::error("Failovers are not implemented yet");
+        throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
+    }
+    else
+    {
+        // Shouldn't get here, would have failed in getFailoverBlockedReason
+        lg2::info("StartFailover on active BMC not supported yet");
+        throw sdbusplus::xyz::openbmc_project::Common::Error::Unavailable();
+    }
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> Manager::doFailoverFromPassive()
+{
+    lg2::info("Starting failover");
+
+    // NOLINTNEXTLINE(clang-analyzer-core.uninitialized.Branch)
+    co_await providers->getSyncInterface().disableBackgroundSync();
+
+    // Stop handling as a passive BMC. This BMC no longer needs to
+    // watch for any changes from the one we're about to reset.
+    handler.reset();
+
+    redundancyInterface.failover_imminent(true);
+
+    // Wait to give the sibling a chance to react to seeing
+    // failover imminent before continuing.
+    co_await providers->getServices().doFailoverImminentDelay();
```

Comment:
I wonder. If this BMC (currently the passive, but in progress to be active) were to get rebooted right here, does our reboot flows handle that? Would the current active just stay the active?